control-plane/roles/isolated-clusters/templates/registry/registry-sts.yaml

```diff
@@ -0,0 +1,54 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: registry
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  labels:
+    app: registry
+  name: registry
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: registry
+  serviceName: registry
+  template:
+    metadata:
+      labels:
+        app: registry
+        name: registry
+    spec:
+      containers:
+      - name: registry
+        image: registry:{{ isolated_clusters_registry_version }}
+        imagePullPolicy: IfNotPresent
+        env:
+        - name: REGISTRY_STORAGE_FILESYSTEM_ROOTDIRECTORY
+          value: "/var/lib/registry"
+        - name: REGISTRY_STORAGE_DELETE_ENABLED
+          value: "true"
+        ports:
+        - containerPort: 5000
+          name: registry
+        volumeMounts:
+        - mountPath: /var/lib/registry
+          name: storage
+          subPath: registry
+      serviceAccount: registry
+      serviceAccountName: registry
+  volumeClaimTemplates:
+  - apiVersion: v1
+    kind: PersistentVolumeClaim
+    metadata:
+      name: storage
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      resources:
+        requests:
+          storage: {{ isolated_clusters_registry_storage_size }}
+      storageClassName: partition-gold
```

Comment:
Should be optional and factored out as this only works for metal-stack shoots.

Reply:
Introduce: `isolated_clusters_registry_storage_class_name`